app/products/calls/actions/calls.ts

```diff
@@ -0,0 +1,261 @@
+                    const muted = call.states && call.states[curIdx] ? !call.states[curIdx].unmuted : true;
+                    const raisedHand = call.states && call.states[curIdx] ? call.states[curIdx].raised_hand : 0;
+                    accum[cur] = {id: cur, muted, raisedHand};
+                    return accum;
+                }, {} as Dictionary<CallParticipant>),
+                channelId: channel.channel_id,
+                startTime: call.start_at,
+                screenOn: call.screen_sharing_id,
+                threadId: call.thread_id,
+            };
+        }
+        enabledChannels[channel.channel_id] = channel.enabled;
+    }
+
+    setCalls(serverUrl, userId, callsResults, enabledChannels);
+
+    return {data: {calls: callsResults, enabled: enabledChannels}};
+};
+
+export const loadConfigAndCalls = async (serverUrl: string, userId: string) => {
+    const res = await checkIsCallsPluginEnabled(serverUrl);
+    if (res.data) {
+        loadConfig(serverUrl, true);
+        loadCalls(serverUrl, userId);
+    }
+};
+
+export const checkIsCallsPluginEnabled = async (serverUrl: string) => {
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    let data: ClientPluginManifest[] = [];
+    try {
+        data = await client.getPluginsManifests();
+    } catch (error) {
+        await forceLogoutIfNecessary(serverUrl, error as ClientError);
+        return {error};
+    }
+
+    const enabled = data.findIndex((m) => m.id === Calls.PluginId) !== -1;
+    setPluginEnabled(serverUrl, enabled);
+
+    return {data: enabled};
+};
+
+export const enableChannelCalls = async (serverUrl: string, channelId: string) => {
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    try {
+        await client.enableChannelCalls(channelId);
+    } catch (error) {
+        await forceLogoutIfNecessary(serverUrl, error as ClientError);
+        return {error};
+    }
+
+    setChannelEnabled(serverUrl, channelId, true);
+    return {};
+};
+
+export const disableChannelCalls = async (serverUrl: string, channelId: string) => {
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    try {
+        await client.disableChannelCalls(channelId);
+    } catch (error) {
+        await forceLogoutIfNecessary(serverUrl, error as ClientError);
+        return {error};
+    }
+
+    setChannelEnabled(serverUrl, channelId, false);
+    return {};
+};
+
+export const joinCall = async (serverUrl: string, channelId: string, intl: IntlShape) => {
+    // Edge case: calls was disabled when app loaded, and then enabled, but app hasn't
+    // reconnected its websocket since then (i.e., hasn't called batchLoadCalls yet)
+    checkIsCallsPluginEnabled(serverUrl);
```

Comment:
Not really--it's there so that the ChannelInfo gets updated in that edge case condition noted in the comments. But your comment made me think: if it's not enabled, we could actually just exit early, since the `newConnection` in line 212 would fail (which is fine, that's what we were relying on before). But this way the intention is a bit clearer for readers of the code. Thanks!